evals/record.py

```diff
@@ -341,6 +342,88 @@ def record_final_report(self, final_report: Any):
         logging.info(f"Final report: {final_report}. Logged to {self.event_file_path}")
 
 
+class HttpRecorder(RecorderBase):
+    def __init__(
+        self, url: str, run_spec: RunSpec, local_fallback_path: str, batch_size: int = 100
+    ):
+        super().__init__(run_spec)
+        self.url = url
+        self.batch_size = batch_size
+        self.failed_requests = 0  # Add this line to track failed requests
+        self.local_fallback_path = local_fallback_path
+        self.local_fallback_recorder = LocalRecorder(local_fallback_path, run_spec)
+        logger.info(f"HttpRecorder initialized with URL {self.url}")
+
+    def _flush_events_internal(self, events_to_write: Sequence[Event]):
+        batch_size = self.batch_size
+        for i in range(0, len(events_to_write), batch_size):
+            batch = list(events_to_write[i : i + batch_size])
+            try:
+                self._send_event(batch)
+            except RuntimeError as e:
+                logger.error(f"Falling back to LocalRecorder due to error: {str(e)}")
+                self.local_fallback_recorder._flush_events_internal(batch)
+                raise RuntimeError(
+                    "An error occurred when sending events."
+                    "Your events have been saved locally using the Local recorder."
+                )
+
+    def _send_event(self, events: List[Event]):
+        # Convert the events to dictionaries
+        events_dict = [dataclasses.asdict(event) for event in events]
+
+        logger.debug(f"Sending events: {events_dict}")
+
+        try:
+            # Send the events to the specified URL
+            response = requests.post(self.url, json=events_dict)
+
+            # If the request succeeded, log a success message
+            if response.ok:
+                logger.debug(f"Events sent successfully")
+
+            # If the request failed, log a warning and increment failed_requests
+            else:
+                logger.warning(f"Failed to send events: {response.text}")
+                self.failed_requests += len(
+                    events
+                )  # Increase the count by the number of events in the failed request
+
+        except Exception as e:
+            logger.warning(f"Failed to send events: {str(e)}")
+            self.failed_requests += len(
+                events
+            )  # Increase the count by the number of events in the failed request
+
+            # Check if the proportion of failed requests exceeds the threshold
+            fail_threshold = 0.05  # Set the threshold to 5%
```

Comment:
Could we make this configurable from the CLI args? 5% default seems good.

Reply:
Hey @jwang47,

Thanks for yet another review.
In my latest commit I am addressing this, in hopes that it suits the need.